.github/workflows/caching.yml

```
@@ -182,25 +184,35 @@ jobs:
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
         run: |
-          cabal update
+          cabal v2-update
+
+      - if: steps.compiled-deps.outputs.cache-hit != 'true' && runner.os == 'Linux' && matrix.ghc == '8.10.7'
+        name: Download sources for bench
+        #  Downloaded separately, to match the tested work/PR workflow guarantees
+        run: |
+          cabal $cabalBuild --enable-benchmarks --only-download
 
       - if: steps.compiled-deps.outputs.cache-hit != 'true'
-        name: Download all sources
+        name: Download the rest of the sources
+        #  Downloaded separately, to match the tested work/PR workflow guarantees
         run: |
-          cabal $cabalBuild --only-download
+          cabal $cabalBuild --enable-tests      --only-download
```

minor whitespace thing
```suggestion
          cabal $cabalBuild --enable-tests --only-download
```